File: app/src/main/java/seigneur/gauvain/mycourt/ui/shots/list/data/NetworkState.java
package seigneur.gauvain.mycourt.ui.shots.list.data;

public class NetworkState {

    private Status status;

    private String message;

    private NetworkState(Status status, String message) {
        this.status = status;
        this.message = message;
    }

    private NetworkState(Status status) {
        this.status = status;
    }

    public static NetworkState LOADED = new NetworkState(Status.SUCCESS);

    public static NetworkState LOADING = new NetworkState(Status.RUNNING);

    public static NetworkState error(String message) {
        return new NetworkState(Status.FAILED, message == null ? "unknown error" : message);
    }

    public Status getStatus() {
        return status;
    }

    public String getMessage() {
        return message;
    }

}

File: app/src/main/java/seigneur/gauvain/mycourt/ui/shots/list/data/NetworkState.kt
package seigneur.gauvain.mycourt.ui.shots.list.data

class NetworkState {

    var status: Status? = null
        private set

    var message: String =""

    private constructor(status: Status, message: String) {
        this.status = status
        this.message = message
    }

    private constructor(status: Status) {
        this.status = status
    }

    companion object {

        var LOADED = NetworkState(Status.SUCCESS)

        var LOADING = NetworkState(Status.RUNNING)

        fun error(message: String?): NetworkState {
            return NetworkState(Status.FAILED, message ?: "unknown error")
        }
    }

}
